src/app/core/services/misc/github-client.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core'
+
+import { GithubContent } from '../../../shared/models/github'
+import { AppServerService } from '../app-server/app-server.service'
+
+@Injectable()
+export class GithubClient {
+  constructor(private appServerService: AppServerService) {}
+
+  getRaw(url): Promise<any> {
+    return this.appServerService.fetchFromGithub(url)
+  }
+
+  getContent(url): Promise<any> {
+    return this.getRaw(url).then((res: GithubContent) => {
+      return JSON.parse(atob(res.content))
```

Comment:
do you need to support proper encoding here too like the protocols?

Reply:
Yeah there are fixes in another PR. I can add the changes once that's merged so that we can reuse the function.

Reply:
Ok it's been merged and I've made the changes.